### PR TITLE
fix inherited variables in pre, post and clean scripts

### PR DESCRIPTION
**Pull request by luispinedamx**
- Also corrected the assignment of job variables to jobInfo.
- Includes functional test.

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java [hunk 1]
```diff
@@ -1282,4 +1282,10 @@ public void setScheduledTimeForRemoval(long scheduledTimeForRemoval) {
         this.jobInfo.setScheduledTimeForRemoval(scheduledTimeForRemoval);
     }
 
+    @Override
+    public void setVariables(Map<String, JobVariable> variables) {
+        super.setVariables(variables);
+        this.jobInfo.setVariables(this.getVariablesAsReplacementMap());
+    }
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java]: why not. I think you should do the same with genericInformation. I tested that the jobInfo contains a null generic info when it should not.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TerminationData.java [hunk 2]
```diff
@@ -309,9 +310,23 @@ public VariablesMap getStringSerializableMap(SchedulingService service, TaskTerm
         } else {
             variablesMap.setInheritedMap(fillMapWithTaskResult(taskResult, true));
         }
+        variablesMap.setScopeMap(removeInheritedVarsFromScope(variablesMap, taskData.getTask().getVariables()));
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TerminationData.java]: it looks like the following line, appearing earlier in the code: variablesMap.setScopeMap(taskData.getTask().getScopeVariables());
is overwritten by your code, so it should be removed

Comment (luispinedamx) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TerminationData.java]: fixed @fviale !

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java (hunk 1 above)
Comment (luispinedamx) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java]: addressed @fviale

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java [hunk 3]
```diff
@@ -1282,4 +1282,16 @@ public void setScheduledTimeForRemoval(long scheduledTimeForRemoval) {
...
+
+    @Override
+    public void setGenericInformation(Map<String, String> genericInformation) {
+        super.setGenericInformation(genericInformation);
+        this.jobInfo.setGenericInformation(genericInformation);
+    }
+
```
Comment (luispinedamx) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java]: Now, both variables and generic information are propagated to all tasks and available through getTaskInfo().getJobInfo().getVariables() (or getGenericInformation()).

**Pull request merged**